src/iter/index.ts

```diff
@@ -100,8 +101,11 @@ export function iter <M extends QueryMethodTypes> (iterResult: IterResult<M>, op
     // Handle frequency and interval
     counterDate.add(options, filtered)
 
-    if (counterDate.year > dateutil.MAXYEAR) {
-      return emitResult(iterResult)
+    if (counterDate) {
+
+      if (counterDate.year > endYear) {
```

Comment:
Please collapse these 2 `if` statements into `if (counterDate && counterDate.year > endYear)`

Reply:
I think you should take it despite that now no?